feat: Implemented bulletin update endpoint

khesir opened the pull request.

miggyalino commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +69,98 @@ export class BulletinService {
+      try {
+        const updatedBulletin = await tx.bulletin.update({
+          where: { id },
+          data: {
+            category_id: updateBulletinDto.category_id,
+            title: updateBulletinDto.title,
+            content: updateBulletinDto.content,
+            author: updateBulletinDto.author,
+          },
```
Use spread operator. No need to specify each attribute here.

`...updateBulletinDto`

khesir replied:
No, I rather not do that, updatebulletinDto is a request payload from frontend and it does not have the id to define the pdf. ID is only define by the http params.

miggyalino commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +69,98 @@ export class BulletinService {
+          },
+        });
+
+        if (pdfAttachments && pdfAttachments.length > 0) {
```
create isEmpty util function

khesir replied:
I feel like this part of the code unecessary to create a util function, because still readable and only a single line of code. Considering this one.

miggyalino commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +69,98 @@ export class BulletinService {
+              },
+            });
+
+            if (references.length === 0) {
```
create isEmpty util function

khesir replied:
I feel like this part of the code unecessary to create a util function, because still readable and only a single line of code. Considering this one.

miggyalino commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +69,98 @@ export class BulletinService {
+          // This create new records
+          for (const pdfAttachment of pdfAttachments) {
+            const now = new Date();
+            const formattedDate = `${String(now.getDate()).padStart(
+              2,
+              '0',
+            )}-${String(now.getMonth() + 1).padStart(
+              2,
+              '0',
+            )}-${now.getFullYear()}`;
+            const sanitizedFileName = pdfAttachment.originalname
+              .replace(/\s+/g, '-')
+              .replace(/[^a-zA-Z0-9.-]/g, '');
+            const uniqueFileName = `${formattedDate}-${sanitizedFileName}`;
+            const filePath = uniqueFileName;
```
Does it have to be in this format? Can't we use date-fns for formatting? @SaaammmyyyS 
If we really have a specified format, we can do a custom util function pero if this is doing normal formatting lang, let's just utilize date-fns

SaaammmyyyS reviewed (commented):
If successful should return the whole updated data

khesir commented on DOCS.md:
```diff
@@ -231,3 +254,72 @@ Specific message appears on which field is mising
+cathegory_id: int, required
+title: string, required
+content: string, required
+author: string, required
```
fff wla na edit hahahah

SaaammmyyyS commented on src/bulletin/bulletin.controller.ts:
```diff
@@ -70,4 +74,49 @@ export class BulletinController {
+  ) {
+    try {
+      await this.bulletinService.updateBulletin(
+        id,
```
Refactore the `updateBulletin` method to pass the entire BulletinDTO

khesir commented:
## Summary
### Changes:
- Change util.service to a stateless function
 - Moved the uploading to pdf file to supabase service
 - Added deleted_attachement_ids to bulletindto
 - Some logic changes on update Bulletin service such as, only deleting selected pdf attachement ids and adding new ones
- Written some documentation to use this endpoint

### Added:
- Created custom util function messageCreate  to conditionally concat messages like `added 1 Pdf attachment` and `deleted 2 attachments`

khesir commented:
Sample sa form data for testing

![image](https://github.com/user-attachments/assets/cf7f15ab-917d-4116-ab5b-a257b919a7d1)

rhaeyx commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +44,79 @@ export class BulletinService {
+
+          await Promise.all(
+            deleted_attachment_ids.map((deleteID) =>
+              tx.pDFAttachment.update({
```
Is it possible to utilize updateMany() dito para di na need mag map through.

khesir replied:
alright I'll do that

khesir replied:
Woops, I tried doing that, but I realize that I need individual values on deleted_Attacment_ids so still need to iterate through all the items in the list.

I just did promise all to do it concurrently, to process it all at once.

SaaammmyyyS commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +44,79 @@ export class BulletinService {
+        }
+        // Check if there is new pdf attachments
+        // Upload all and create new data for pdfAttacment table
+        if (!isEmpty(pdfAttachments)) {
```
same here, Instead of creating PDF attachments one by one, you can use `createMany`

khesir replied:
nah impossible to do that, I process both uploading to storage and creating the data sa pdfattachment.

I need to confirm na ma upload ang pdf sa db then I create pdfattachment data.

I can separate the upload data sa pdf then create many. but thats another process naput

khesir commented on src/bulletin/bulletin.service.ts:
```diff
@@ -69,4 +44,79 @@ export class BulletinService {
+  ) {
+    return this.prismaService.$transaction(async (tx) => {
+      try {
+        const { deleted_attachment_ids, ...bulletinData } = updateBulletinDto;
```
If i edit nako mag change ang dto to deletedAttachmentIds, kay gina get nako ang request body { delete_attachment_id }
